src/driver/driver.rs

```diff
@@ -0,0 +1,45 @@
+//! The driver trait which all drivers will implement.
+
+use schema::Definition;
+use error::{Error, ErrorCode};
+use patch::Patch;
+use value::{Pointer, Value};
+use query::Query;
+
+pub trait Driver {
+  /// Connects to a driver and returns a driver instance. After calling this
+  /// the driver is ready to roll!
+  fn connect(uri: &str) -> Result<Box<Self>, Error>;
```

Comment:
Actually change that, it will definitely break things. Self will be defined as the implementing type when the specializations are created therefore it will end up as `Box<MyDriver>` instead of what we want: `Box<Driver>`